src/d3a/d3a_core/simulation.py

```diff
@@ -256,7 +255,9 @@ def _run_cli_execute_cycle(self, slot_resume, tick_resume):
 
     def _update_and_send_results(self, is_final=False):
         self.endpoint_buffer.update_stats(self.area, self.status, self.progress_info)
-        if self.export_on_finish and self.should_export_results and is_final:
+        import d3a.constants
+        if self.export_on_finish and self.should_export_results and \
+                self.area.current_market is not None and d3a.constants.D3A_TEST_RUN:
```

Comment:
Here, too: Change to `from d3a.constants import D3A_TEST_RUN` and move to top.

Reply:
https://github.com/gridsingularity/d3a/pull/868#discussion_r490020878